app/javascript/components/student/Editor.tsx

```diff
@@ -1,34 +1,102 @@
 import React, { useReducer, useRef, useEffect, useCallback } from 'react'
-import { CodeEditor } from './editor/CodeEditor'
 import { TestRunSummary } from './editor/TestRunSummary'
 import { Submitting } from './editor/Submitting'
 import { fetchJSON } from '../../utils/fetch-json'
 import { typecheck } from '../../utils/typecheck'
+import MonacoEditor from 'react-monaco-editor'
+import * as monacoEditor from 'monaco-editor/esm/vs/editor/editor.api'
 
-export type Submission = {
-  testsStatus: TestRunStatus
-  uuid: string
-  links: SubmissionLinks
+export type Submission =
```

Comment:
Our submission JSON now includes the test_run of the submission. This allows us to just pass in a submission object initially, and the test_run will be populated.

This type definition outlines all the possible test statuses, and whether the specific test status contains a test run or not.